java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -91,6 +91,7 @@ public ColumnView(DType type, long rows, Optional<Long> nullCount,
    * @param validityBuffer an optional validity buffer. Must be provided if nullCount != 0.
    *                       The ownership doesn't change on this buffer
    */
+  @Deprecated
```

Comment:
Why is this deprecated?  You can create most column views in a perfectly valid way with this?  Just not nested ones.

Reply:
I deprecated this because 1. It doesn't take in an offset buffer which is needed to create Strings and 2. Because ColumnVector doesn't have a similar API. But I do agree this is a much simpler API and makes more sense. We should change the way we expect the users of Java API to think of Strings as columns with no children.